### ISSUE TITLE
fman 1.1.9: Can't move files

### ISSUE THREAD

**Issue by raguay** [fman-community/fman#485]
I'm trying to move files to a subfolder (The message is for moving just one), but I keep getting the following message:

Command 'Move' raised error.

Traceback (most recent call last):
  File "fman/impl/fs_cache.py", line 36, in query
KeyError: 'stat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "fman/impl/plugins/command_registry.py", line 110, in _execute_command
  File "/Applications/fman.app/Contents/Resources/Plugins/Core/core/commands/__init__.py", line 479, in __call__
    self._call(files, dest_dir, src_dir, dest_name)
  File "/Applications/fman.app/Contents/Resources/Plugins/Core/core/commands/__init__.py", line 637, in _call
    submit_task(MoveFiles(files, dest_dir, src_dir, dest_name))
  File "fman/__init__.py", line 319, in submit_task
  File "/Applications/fman.app/Contents/Resources/Plugins/Core/core/fileoperations.py", line 41, in __call__
    if not self._gather_files():
  File "/Applications/fman.app/Contents/Resources/Plugins/Core/core/fileoperations.py", line 101, in _gather_files
    self._enqueue(self._prepare_transfer(src, dest))
  File "/Applications/fman.app/Contents/Resources/Plugins/Core/core/fileoperations.py", line 155, in _enqueue
    for task in self._iter(tasks):
  File "/Applications/fman.app/Contents/Resources/Plugins/Core/core/fileoperations.py", line 211, in _iter
    for item in iterable:
  File "/Applications/fman.app/Contents/Resources/Plugins/Core/core/fs/local/__init__.py", line 87, in _prepare_move
    dst_dir_dev = self.stat(splitscheme(dirname(dst_url))[1]).st_dev
  File "fman/fs.py", line 201, in wrapper
  File "fman/impl/fs_cache.py", line 12, in query
  File "fman/impl/fs_cache.py", line 38, in query
  File "fman/fs.py", line 201, in <lambda>
  File "/Applications/fman.app/Contents/Resources/Plugins/Core/core/fs/local/__init__.py", line 44, in stat
    return os.stat(self._url_to_os_path(path))
FileNotFoundError: [Errno 2] No such file or directory: 'Users/raguay/Envato/CreateYourOwnAlexaSkill/orig/Users/raguay/Envato/CreateYourOwnAlexaSkill/orig'

It's copying the directory structure twice for the path to move the file to. This is just since the 1.1.9 update.

**Comment by raguay** [comment]
I just noticed that the path in the bar at the top of the panel was missing the beginning '/'. I added it and copy/moving now works. fman needs to ensure that the path in the top bar is complete. This also fixed the issue with the dropboxconnect plugin.

**Comment by mherrmann** [comment]
Interesting. How did you get the path in the location bar without a leading `/` in the first place? (And sorry, again, for the trouble.)

**Comment by raguay** [comment]
Not sure. It launched after the update that way. I didn't notice it at
first until I couldn't copy/move anything. Then I happened to notice the
leading slash was missing. When I added it, everything started working.
Therefore, something about the update had it loose it, but it hasn't
happened again since then. Kind of strange.

**Comment by mherrmann** [comment]
Strange :/ If it does happen again or you can see a pattern when/why it does, would be great if you could let me know.

**Comment by raguay** [comment]
👍🏻

**Comment by rmgk** [comment]
1. Enter the name of a folder in your home directory (need to enter exactly as is, not fuzzy matching)
![bildschirmfoto zu 2018-08-21 11-36-25](https://user-images.githubusercontent.com/151862/44394441-7e90c000-a537-11e8-9fb3-41da60f10b80.png)

2. Notice relative path in top bar.
![bildschirmfoto zu 2018-08-21 11-36-57](https://user-images.githubusercontent.com/151862/44394482-95cfad80-a537-11e8-8fb4-82fe98d5a8f8.png)

3. fman is now in confused mode – using the pane with the relative path causes all sorts of operations to fail, and GoTo also remembers the relative path
![bildschirmfoto zu 2018-08-21 11-37-16](https://user-images.githubusercontent.com/151862/44394497-a1bb6f80-a537-11e8-84b6-4505787e1a72.png)

**Comment by mherrmann** [comment]
Damn. Will release a fix for this today.

**Comment by raguay** [comment]
Yep, that does do it, but that isn't how I came across it.

As I said, the top bar needs to make sure it is always the proper absolute
path for each OS. Even when typing in, the user input needs to be checked!

**Comment by rmgk** [comment]
@raguay but is it completely unrelated? Clicking the top bar also just opens the goto dialog, is there even any other way to input the path?

There seem to be some more issues (maybe related issues) with GoTo:
1. If you enter gibberish (no more selections available) and hit enter you get an exception:
```
Command 'GoTo' raised error.

Traceback (most recent call last):
  File "fman/impl/plugins/command_registry.py", line 110, in _execute_command
  File "/opt/fman/Plugins/Core/core/commands/__init__.py", line 1000, in __call__
    url = self._get_target_location(*result)
  File "/opt/fman/Plugins/Core/core/commands/__init__.py", line 1022, in _get_target_location
    return as_url(suggested_dir)
  File "fman/url.py", line 22, in as_url
  File "pathlib.py", line 622, in __new__
  File "pathlib.py", line 651, in _from_parts
  File "pathlib.py", line 643, in _parse_args
TypeError: argument should be a path or str object, not <class 'NoneType'>
```

2. If you enter a path, say `/home/ragnar` (e.g., by clicking the address bar) and then use the keyboard or mouse to select one of the entries, e.g., `/home/ragnar/Downloads`, you will go to the directory you have entered (i.e., `/home/ragnar`) not the one you selected. Selection does work, if the string you entered is not a valid path.

**Comment by raguay** [comment]
I didn't say it's not related. That's just not how I came across it. For me, after the update, all of the top bars had the leading '/' missing. The results are the same as you are describing here.

The fix: check for a proper absolute path anytime the goto functionality is used.

**Comment by mherrmann** [comment]
> The fix: check for a proper absolute path anytime the goto functionality is used.

Implemented and released this now as [fman 1.2.0](https://fman.io/changelog#v1.2.0). Thank you both for reporting and helping with the troubleshooting.

**Comment by raguay** [comment]
No problem. Glad to help.
